NeoML/include/NeoML/Dnn/DnnHead.h

```diff
@@ -64,13 +64,30 @@ struct CDnnHead : public IObject {
 	{
 		++headCounter;
 		if(headCounter == connections.Size()) {
+			innerInputBlobs.DeleteAll();
+			innerOutputBlobs.DeleteAll();
 			headCounter = 0;
 			firstAdapterNum = -1;
 		}
 	}
 
 	CDnn* dnn;
+	// Stores all adapter using this head
 	CObjectArray<CDnnHeadAdapterLayer> connections;
+	// Layers for which input/output blobs are stored for Backward/Learn
+	CArray<CBaseLayer*> inputLayers;
+	CArray<CBaseLayer*> outputLayers;
+	// Pointers to source/sink layers of inner network
+	CCompositeSourceLayer* sourceLayer = nullptr;
+	CCompositeSinkLayer* sinkLayer = nullptr;
+	// Stores the input/output blobs from last Inference
+	CObjectArray<CDnnBlob> innerInputBlobs;
```

Comment:
But `innerInputBlobs` and `innerOutputBlobs` are differs for all adapters, it may be left in layer class